engineio/context.go

```diff
@@ -0,0 +1,29 @@
+package engineio
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/googollee/go-socket.io/engineio/packet"
+)
+
+// Context provides info for a HTTP request.
+type Context struct {
+	// Standard context
+	context.Context
+
+	// The instance of current session.
+	Session Session
+
+	// The http.Request of current HTTP request.
+	// It may be different when using polling transport.
+	Request *http.Request
+
+	// The current packet infomation.
+	PacketType packet.Type
+	Reader     io.Reader
```

Comment:
Googol, what do you think about "hide" this interface?
i mean hide as allow to call `Read()` method from context directly. e.g. `ctx.Read()`

Reply:
No. I don't think it's good to make a Context as a Reader. But I think it should be `Body`.

Reply:
Made a packet struct in Context, and rename the reader as Body.